microprofile/WFLY-12590_Integrate_MicroProfile_Fault_Tolerance_2.0.adoc

```diff
@@ -0,0 +1,102 @@
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mnovak@redhat.com[Miroslav Novak]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+
+* The specification repository resides at official Eclipse organization on GitHub in https://github.com/eclipse/microprofile-fault-tolerance[eclipse/microprofile-fault-tolerance] repository.
+* The implementation of Eclipse MicroProfile Fault Tolerance will be provided by SmallRye project https://github.com/smallrye/smallrye-fault-tolerance[smallrye/smallrye-fault-tolerance].
+* The underlying library used by _current_ SmallRye implementation is Hystrix which resides in official Netflix GitHub organization https://github.com/Netflix/Hystrix[Netflix/Hystrix].
+
+=== Other Interested Projects
+
+Not interested in our integration per se, but projects that have already integrated MP FT:
+
+* https://thorntail.io/[Thorntail] has integrated MicroProfile Fault Tolerance as a https://github.com/thorntail/thorntail/tree/master/fractions/microprofile/microprofile-fault-tolerance[fraction].
+* https://quarkus.io/[Quarkus] has integrated MicroProfile Fault Tolerance as an https://github.com/quarkusio/quarkus/tree/master/extensions/smallrye-fault-tolerance[extension].
+
+== Requirements
+
+* Support Eclipse MicroProfile Fault Tolerance version 2.0 for all applications deployed in WildFly.
+* [WILL BE CLARIFIED] The extension is either added to all configurations provided by WildFly (standalone and domain mode)
+  or provided as a layer in a separate build profile.
+* Upstream Eclipse MicroProfile Fault Tolerance TCK test suite will be added to `microprofile-tck` WildFly upstream test suite.
+
+== Nice-to-Have Requirements
+
+* [WILL BE CLARIFIED] Create a MicroProfile Fault Tolerance https://github.com/wildfly/quickstart[quickstart].
+
+Items that are currently only being taken into consideration with next version of this feature:
+
+* The underlying library used by SmallRye FT implementation is Hystrix which is currently in "maintenance" mode and a final
+  release of the project (version 1.5.18) has been released. The discussion about replacing the library is ongoing.
+* WildFly's standard management model will be configuration-less
+** all configuration is done through MicroProfile Config specification
+** given the maintenance problem above it is not desirable to expose a management model which will change in near future as concepts are not cleanly mappable to a different underlying library
+* Support for plurality of isolated deployments.
+
+== Non-Requirements
+
+* N/A
+
+== Implementation Plan
+
+* Add dependencies on `smallrye-fault-tolerance` and `microprofile-fault-tolerance-api` artifacts.
```

Comment:
I assume 2 new modules. Am I correct ?

Reply:
From the deployment dependencies perspective that's correct. From perspective of the application server there are many new modules (hystrix, commons-config, archies, rxjava v1, etc).